### ISSUE TITLE
NIOWebSocketServerUpgrader lost head of WebSocket stream

### ISSUE THREAD

**Issue by omochi** [apple/swift-nio#1003]
### Expected behavior
NIOWebSocketServerUpgrader handle TCP stream after negotiation HTTP Request treat WebSocket stream.

### Actual behavior
TCP stream received at server before upgrade completion will lost.

### Steps to reproduce

1. Add this logging code.

```swift
extension ByteToMessageHandler: ChannelInboundHandler {
    public func handlerRemoved(context: ChannelHandlerContext) {
        let buffer = cumulationBuffer
        let rem = buffer?.readableBytes ?? 0
        print("ByteToMessageHandler handlerRemoved: rem=\(rem)")
        ...
    }
}
```

2. Run NIOWebSocketServer
3. Send request and websocket stream immediately.

I test it with this.

```
$ cat a.txt
GET / HTTP/1.1
Host: localhost:8888
Upgrade: websocket
Connection: upgrade
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: E4WSEcseoWr4csPLS2QJHA==

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
$ cat a.txt | nc localhost 8888
```

4. In console:
```
ByteToMessageHandler handlerRemoved: rem=64
```

### Detail

`configureHTTPServerPipeline` method insert `ByteToMessageHandler<HTTPRequestDecoder>` front of `HTTPServerUpgradeHandler`.
So head of websocket stream is received by it.
But it may not shape HTTP Request header, so these stream is buffered.
When upgrade happen, these data will lost.

### Related

I first faced this issue at client side.
This is not in swift-nio, vapor.
https://github.com/vapor/nio-websocket-client/issues/1
I was trouble in this problem, to search solution, I read `NIOWebSocketServer` in swift-nio.
And I found same issue at here.

### SwiftNIO version/commit hash

`2.1.0` `0e37ada7cc6b68452716d706849b6868a2a77429`

### Swift & OS version (output of `swift --version && uname -a`)

```
Apple Swift version 5.0.1 (swiftlang-1001.0.82.4 clang-1001.0.46.5)
Target: x86_64-apple-darwin18.5.0
Darwin omochi-iMacPro.local 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_64
```

**Comment by Lukasa** [comment]
This data should not be lost if you use `configureHTTPServerPipeline`, as it configures the `RemoveAfterUpgradeStrategy` of `HTTPDecoder` to `forwardBytes`, so the bytes should not be dropped.

However, this fix is only present in NIO 2, which is not currently used by Vapor.

**Comment by omochi** [comment]
Thanks to response immediately!
And I am sorry I did not noticed about `forwardBytes`, I confirmed that `context.fireChannelRead(NIOAny(buffer))` is called in `decodeLast` method.

Its okey. nio-websocket-client uses swift-nio 2.0.0.
https://github.com/vapor/nio-websocket-client/blob/master/Package.swift
And it already use `leftOverBytesStrategy`.
https://github.com/vapor/nio-websocket-client/blob/master/Sources/NIOWebSocketClient/WebSocketClient.swift#L68
I think I can fix problem in client if I implement same logic as this which transfer such bytes 
https://github.com/apple/swift-nio/blob/993a3b7521fa636cd8894c3c053bc93c7394c17d/Sources/NIOHTTP1/HTTPServerUpgradeHandler.swift#L226-L235

Thank you very much.
